toolkit/tools/imagecustomizer/container/run.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+# The first argument is expected to be a version tag like '2.0.20240615',
+# '2.0.latest', '3.0.20240615-rc', etc.
+VERSION_TAG=$1
+
+# Construct the full OCI artifact path based on the provided version tag. Below
+# splits the VERSION_TAG into major version (e.g., '2.0') and the specific tag
+# (e.g., '20240615' or 'latest').
+MAJOR_VERSION=$(echo $VERSION_TAG | cut -d'.' -f1-2)
+SPECIFIC_TAG=$(echo $VERSION_TAG | cut -d'.' -f3-)
+
+# Check if the specific tag is 'latest'. If not, prepend the MAJOR_VERSION to
+# it.
+if [ "$SPECIFIC_TAG" != "latest" ]; then
+    SPECIFIC_TAG="${MAJOR_VERSION}.${SPECIFIC_TAG}"
+fi
+
+# Construct the OCI Artifact full path.
+OCI_ARTIFACT_PATH="mcr.microsoft.com/azurelinux/${MAJOR_VERSION}/image/minimal-os:${SPECIFIC_TAG}"
+
+ARTIFACT_DIR="/oci/artifact"
+mkdir -p $ARTIFACT_DIR
+oras pull $OCI_ARTIFACT_PATH -o $ARTIFACT_DIR
```

Comment:
should we check the exit code of `oras pull` and error out that version is not found instead of the vhdx error below, it can be confusing.

Reply:
I have updated the error message, but I can't put version not found specifically, there might be some other error reasons like network issue.

Reply:
If we are trying to hide the oci url then maybe not call out explicitly and keep it generic.
For example, "Failed to fetch image with the provided version $version"

Reply:
I see, fixed.